README.md

```diff
@@ -1,2 +1,127 @@
-# nf-synindex
-Nextflow pipeline for indexing S3 objects in Synapse
+# nf-synindex: Index S3 Objects in Synapse
+
+## Purpose
+
+The purpose of this Nextflow workflow is to automate the process of indexing S3 objects in Synapse. These S3 objects are typically the output files from a general-purpose (_e.g._ nf-core) workflow that doesn't contain Synapse-specific steps for uploading or indexing outputs. This workflow is intended to be run after other data processing workflows.
+
+The benefits of using this workflows include:
+
+- There is a lack of data transfer and duplication because the S3 objects remain where they are.
+- The folder-like structure in S3 is reproduced in Synapse to maintain the file organization.
+
+Briefly, `nf-synindex` achieves this automation as follows:
+
+1. Update the `owner.txt` file with the authenticated user ID
+2. Register the Tower bucket as an external storage location
+3. Mirror the directory-like structure in S3 as folders on Synapse
+4. For each S3 object, the following steps are performed in parallel:
+   1. Compute the MD5 checksum
+   2. Register the S3 object as a file handle
+   3. Expose the file handle as a Synapse file under the corresponding Synapse folder
+5. Output a mapping between the S3 objects and the newly created Synapse files
+
+**N.B.** The workflow assumes that the S3 bucket is [configured](https://help.synapse.org/docs/Custom-Storage-Locations.2048327803.html) for Synapse.
+
+## Quickstart
+
+The examples below demonstrate how you would index objects under an S3 prefix in a bucket called `example-bucket`.
+
+1. Identify an S3 prefix that contains a set of objects that you want to index in Synapse (_e.g._ output files from a data processing workflow).
+
+    **Example:** Objects under `s3://example-bucket/outputs/`
+
+    ```text
+    s3://example-bucket/outputs/foobar_1.trimming_report.txt
+    s3://example-bucket/outputs/foobar_2.trimming_report.txt
+    s3://example-bucket/outputs/fastqc/
+    s3://example-bucket/outputs/fastqc/foobar_1_val_1_fastqc.html
+    s3://example-bucket/outputs/fastqc/foobar_1_val_1_fastqc.zip
+    s3://example-bucket/outputs/fastqc/foobar_2_val_2_fastqc.html
+    s3://example-bucket/outputs/fastqc/foobar_2_val_2_fastqc.zip
+    ```
+
+2. Prepare your Synapse configuration file to authenticate the workflow. For more details, check out the [Authentication](#authentication) section.
```

Comment:
This is pending the secrets directive correct?  Should we try to push people to use nextflow secrets over the config file?

Reply:
I plan on elaborating once we have deployed the secrets feature in Tower-Prod.